### PR TITLE
Adds metered message handler.

### PR DESCRIPTION
**Pull request by mikaellothman**
Adds a metered message handler that can be used to push metrics to
a MetricsRegistry.

Ping @iZettle/platform

### CODE REVIEW

File: izettle-messaging/src/main/java/com/izettle/messaging/handler/MeteredMessageHandler.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    public void handle(M message) throws Exception {
+
+        Timer.Context timerContext = timer.time();
+        try {
+            meterAll.mark();
+            actualHandler.handle(message);
+        } catch (RetryableMessageHandlerException e) {
+            meterRetryable.mark();
+        } catch (Exception e) {
+            meterError.mark();
```
Comment (xiaodong-izettle) [inline izettle-messaging/src/main/java/com/izettle/messaging/handler/MeteredMessageHandler.java]: rethrow the exception? :)

File: izettle-messaging/src/main/java/com/izettle/messaging/handler/MeteredMessageHandler.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    @Override
+    public void handle(M message) throws Exception {
+
+        Timer.Context timerContext = timer.time();
+        try {
+            meterAll.mark();
+            actualHandler.handle(message);
+        } catch (RetryableMessageHandlerException e) {
+            meterRetryable.mark();
```
Comment (xiaodong-izettle) [inline izettle-messaging/src/main/java/com/izettle/messaging/handler/MeteredMessageHandler.java]: rethrow RetryableMessageHandlerException? :)

Comment (mikaellothman) [inline izettle-messaging/src/main/java/com/izettle/messaging/handler/MeteredMessageHandler.java]: Oops...nice catch @xiaodong-izettle :)

**Comment by mikaellothman** [comment]
Fixed!

**Comment by xiaodong-izettle** [comment]
I assume you and @nzroller have reached consensus about the naming convention, yes? :)

**Comment by mikaellothman** [comment]
Well, @nzroller can you please have a look?

**Comment by nzroller** [comment]
lookin' now

**Comment by mikaellothman** [comment]
:eyes:

**Comment by nzroller** [comment]
:eyes: .-)

**Pull request merged**